3-UsingAPIs/Week2/assignment/ex3-rollAnAce.js

```diff
@@ -17,21 +17,27 @@ import { rollDie } from '../../helpers/pokerDiceRoller.js';
  * @param {DieFace} desiredValue
  * @returns {Promise<DieFace>}
  */
-export function rollDieUntil(desiredValue) {
-  // TODO rewrite this function using async/await
-  return rollDie().then((value) => {
-    if (value !== desiredValue) {
-      return rollDieUntil(desiredValue);
+export async function rollDieUntil(desiredValue) {
+  try{
+    let value;
+    while (value !== desiredValue) {
+      value = await rollDie();
     }
     return value;
-  });
+  }
+  catch {
+    return console.error(err); 
+  };
```

Comment:
1. Your code contains a bug: the variable `err` is undefined. You can correct it like this:

```suggestion
  catch(err) {
    return console.error(err); 
  };
```

But if you do, there is another bug: You should not catch the error here. If you do, the `rollDieUntil()` function will return a promise that resolves to `undefined` instead of a rejected promise and you will see the message:

```text
YAY! WE GET EXPECTED RESULT AND IT IS: undefined
```